delay_distributions.Rmd

```diff
@@ -0,0 +1,206 @@
+  ylab("Number of infections")
+```
+
+We would now like to simulate hospitalisations arising from this outbreak.
+We assume that the incubation period (or time from infection to symptom onset) is gamma-distributed with shape 5 and rate 1.
+We further assume that the onset-to-hospitalisation period is lognormally distributed with meanlog 1 and sdlog 1.
+We lastly assume that all infections cause symptoms, and that 30% of symptomatic cases become hospitalised.
+
+**Take 10 minutes** to add onset and hospitalisation dates to the infection data.
+
+```{r inf_hosp}
+library("dplyr")
+## assume a start date, e.g. 10 May of this year
+start_date <- as.Date("2024-05-10")
+## create a new data frame based on the `infection_times` data frame
+## representing for each infected individual
+## - `incubation_period`: the time from infection to symptom onset
+## - `onset_to_hosp`: the time from symptom onset to hospitalisation
+## - `infection_date`: the date on which an individual became infected
+## - `onset_date`: the date on which an individual started showing symptoms
+## - `hosp_date`: the date on which an individual became hospitalised (or NA if
+##    they didn't get hospitalied)
+df <- NULL ## replace NULL with your code
+```
+
+```{r inf_hosp_solution}
+source(here::here("scripts", "snippets", "delay-distributions.r"))
+```
+
+If you have trouble filling any of the empty bits in, have a look at our [solution](delay_distributions_solutions.html#simulating-delayed-epidemiological-data) for reference and copy/paste the code from there.
+
+Now we can plot infections, hospitalisations and onsets. To do so we first convert our data frame to long format.
+
+```r
+df <- hosp_scaled |>
+  pivot_longer(
+    cols = c(infection_date, onset_date, hosp_date),
+    names_to = "type", values_to = "date"
+  )
+ggplot(df, aes(x = date)) +
+  geom_histogram(position = "dodge") +
+  facet_wrap(~ type, ncol = 1) +
+  xlab("Date") +
+  ylab("Count")
+```
+
+# Estimating delay distributions
+
+As mentioned above, our data set of infection, onset and hospitalisation dates is not the typical data set we encounter in outbreaks.
+In reality, we don't ahve infection dates, and we also have to deal with missing data, incomplete observations, data entry errors etc.
+For now, let us just assume we have a data set of symptom onsets and some hospitalisation, and we would like to estimate how long it takes for people to become hospitalised after becoming symptomatic.
+This is an important delay to know about, for example when modelling and forecasting hospitalisations, or more generally for estimating required hospital capacity.
+
+As with all our estimation tasks throughout this course, we will be using stan, a fully-fledged toolkit for estimation.
+We will try to represent the model we simulated above as much as possible in this.
+One way that this could look, for examle, is:
+
+```{r delay_model}
+library("cmdstanr")
+
+delay_model <- '
+  data {
```

Comment:
An option here: https://mc-stan.org/cmdstanr/articles/r-markdown.html

Reply:
That's nice. It won't work if people prefer to copy & paste.